contrib/haproxy/patches/use_env_flags.patch

```diff
@@ -0,0 +1,20 @@
+diff --git Makefile Makefile
+index a66c9b69c..f8a542d47 100644
+--- Makefile
++++ Makefile
+@@ -289,13 +289,13 @@ ARCH_FLAGS        = $(ARCH_FLAGS.$(ARCH))
+ # These CFLAGS contain general optimization options, CPU-specific optimizations
+ # and debug flags. They may be overridden by some distributions which prefer to
+ # set all of them at once instead of playing with the CPU and DEBUG variables.
+-CFLAGS = $(ARCH_FLAGS) $(CPU_CFLAGS) $(DEBUG_CFLAGS) $(SPEC_CFLAGS)
++CFLAGS := $(DEBUG_CFLAGS) $(SPEC_CFLAGS) $(CFLAGS)
+ 
+ #### Common LDFLAGS
+ # These LDFLAGS are used as the first "ld" options, regardless of any library
+ # path or any other option. They may be changed to add any linker-specific
+ # option at the beginning of the ld command line.
+-LDFLAGS = $(ARCH_FLAGS) -g
++LDFLAGS := -g $(LDFLAGS)
```

Comment:
this is wrong; -g should not be added (instead, CFLAGS should be passed together with LDFLAGS)